logging.md

```diff
@@ -0,0 +1,34 @@
+# Conventions for LOGGING
+[See issues](https://github.com/OCR-D/spec/issues/11)
+All logging may be stored in files. 
+If so, they must be given in Provenance as a result. 
+
+## Log Levels
+A more detailed description will be found [here](https://stackoverflow.com/questions/2031163/when-to-use-the-different-log-levels/5278006#5278006)
+### TRACE / ALL
+This is the most verbose logging level to trace the path of the algorithm.
+E.g.: Start/End/Duration of a method, even loops may be logged. 
+Note that logging within loops can dramatically affect performance.
+### DEBUG
+This level is used if parameters and or the status of an algorithm/method should be logged. 
+### INFO
+Log information about used settings on application level.
+### WARN
+This level is used to log information about missing/wrong configurations which may lead to errors.
+### ERROR
+Log all events that produce no or a wrong result. 
+It is useful to output all the information that helps to determine the cause without the need for further investigation.
+
+
+## Format
+The format of the logging output has to be formatted like this:
+TIMESTAMP LEVEL LOGGERNAME - MESSAGE
+***Example:***
+08:03:40.017 WARN edu.kit.ocrd.MyTestClass - A warn message
+:warning: 
+Since only the time stamp is logged, the log should be written to files that change daily.
+
+## METS
+### File Group
+File groups holding logging have to start with prefix "LOG-"
```

Comment:
So in terms of BagIt, you see log data as part of the payload? Provenance metadata, parametrization etc. also part of the payload? How to map between provenance, log data, processes and files?

Reply:
At least the provenance should be part of the payload. The provenance refers to the input/output files (via ID?), which can then be used for assignment. All provenances must then be combined to form a graph.

Reply:
> At least the provenance should be part of the payload.

Still don't understand why. Provenance is by definition metadata and not about the thing but about the process that made the thing. Why should this be different from logging data e.g.?

My main problem here is that METS consists (for our purposes!) of files representing pages in file groups representing different manifestations of these pages. The file groups are the output of processes but they don't reflect processes directly.

Provenance is based on processes which can have arbitrary input/output. With special filegroups that don't reflect pages anymore, this consistency breaks. If provenance was to be tracked in METS (which I would not advise), I see these options:

1) Have a file group `X-PROV` for every file group `X` which a `mets:file` for every `mets:file` in `X` that points to the URL of the process that generated that file
2) Add a custom attribute or (re-/ab-)use an existing attritute on the file itself
3) Have some sort of mapping structure that maps XML IDs to the URL of the process that created them

The last one would at least be general enough to also handle cases like "which process created the table of contents". But it will still be a convention not a standard, extend the size of METS forever and have little use for the institutions (who have their own digitization workflow engines which have their own naming schemes and conventions for provenance).

@cneud @wrznr

Reply:
Provenance is important because it is the only way to reproduce data, but reproducibility is an essential characteristic of scientific activity.
By (our) definition, all files contained in the bagit container must also be referenced in METS. As a result, the provenance must also be referenced in METS if it is part of the payload.
Optionally, we can define it as metadata that should be stored in a tag directory of the bagit container, but then the coupling is relatively loose.  (See comment below)
Also the bagit import/export to/from workspace will become more difficult.